Tasks/DockerV2/dockerpush.ts

```diff
@@ -79,3 +99,58 @@ export function run(connection: ContainerConnection, outputUpdate: (data: string
 
     return promise;
 }
+
+async function publishToImageMetadataStore(connection: ContainerConnection, imageName: string, tags: string[], digest: string, dockerFilePath: string): Promise<any> {
+    const imageUri: string = getResourceName(imageName, digest);
+    const dockerFileContent: string = fs.readFileSync(dockerFilePath, 'utf-8').toString();
+    const baseImageName = getBaseImageName(dockerFileContent);
+    const layers = dockerCommandUtils.getLayers(connection, imageUri);
+    const buildId = parseInt(tl.getVariable("Build.BuildId"));
+    const buildDefinitionName = tl.getVariable("Build.DefinitionName");
+    const buildVersion = tl.getVariable("Build.BuildNumber");
+    const buildDefinitionId = tl.getVariable("System.DefinitionId");
+
+    const requestUrl: string = tl.getVariable("System.TeamFoundationCollectionUri") + "/" + tl.getVariable("System.TeamProject") + "/_apis/deployment/imagedetails?api-version=5.0-preview.1";
+    const requestBody: string = JSON.stringify(
```

Comment:
Move code below this line to a separate method.

Reply:
Refactored the code